changefeedccl: support envelope=enriched for avro

asg0451 opened the pull request:
Add support for the enriched envelope type to avro.

Epic: CRDB-8665
Fixes: #139657

Release note (enterprise change): Add support for
the enriched envelope type to avro.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/140525)
<!-- Reviewable:end -->

asg0451 committed:
changefeedccl: support envelope=enriched for avro

Add support for the enriched envelope type to avro.

Fixes: #139657

Release note (enterprise change): Add support for
the enriched envelope type to avro.

aerfrei commented on pkg/ccl/changefeedccl/encoder_avro.go:
```diff
@@ -256,19 +259,42 @@ func (e *confluentAvroEncoder) EncodeValue(
+			if sourceDataSchema, err = avro.NewFunctionalRecord("source", e.schemaPrefix, fields, sourceFn); err != nil {
+				return nil, err
+			}
+		// key_only handled above, and row is not supported in avro
```
Where is key_only handled above?

asg0451 replied:
on line 207, the first in this method

asg0451 commented on pkg/ccl/changefeedccl/avro/avro.go:
```diff
@@ -55,33 +55,33 @@ import (
-// schemaType is one of the set of avro primitive types.
-type schemaType interface{}
+// SchemaType is one of the set of avro primitive types.
+type SchemaType interface{}
```
They're capitalized to allow them to be used in encoder_avro.go

asg0451 commented:
bors r=aerfrei
